device/core/src/module_client.ts

```diff
@@ -222,4 +223,90 @@ export class ModuleClient extends EventEmitter {
   static fromAuthenticationProvider(authenticationProvider: AuthenticationProvider, transportCtor: any): ModuleClient {
     return InternalClient.fromAuthenticationProvider(authenticationProvider, transportCtor, ModuleClient) as ModuleClient;
   }
+
+  static validateEnvironment(): void {
+    // Codes_SRS_NODE_MODULE_CLIENT_13_029: [ If environment variables EdgeHubConnectionString and IotHubConnectionString do not exist then the following environment variables must be defined: IOTEDGE_WORKLOADURI, IOTEDGE_DEVICEID, IOTEDGE_MODULEID, IOTEDGE_IOTHUBHOSTNAME, IOTEDGE_AUTHSCHEME and IOTEDGE_MODULEGENERATIONID. ]
+
+    const keys = [
+      'IOTEDGE_WORKLOADURI',
+      'IOTEDGE_DEVICEID',
+      'IOTEDGE_MODULEID',
+      'IOTEDGE_IOTHUBHOSTNAME',
+      'IOTEDGE_AUTHSCHEME',
+      'IOTEDGE_MODULEGENERATIONID'
+    ];
+
+    keys.forEach((key) => {
+      if (!process.env[key]) {
+        throw new errors.PreconditionFailedError(
```

Comment:
>PreconditionFailedError [](start = 25, length = 23)

that's an error that is usually used for service related errors (especially invalid etags). I'd say this ought to be either a ReferenceError (because the key isn't found) or an ArgumentError (because that's what we throw when the connection string is invalid)